src/main/java/seedu/address/model/tag/Tag.java

```diff
@@ -32,6 +32,39 @@ public static boolean isValidTagName(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    /**
+     * Returns true if another tag has the same TagName as this tag.
+     * @param otherTag A tag to compare with.
+     */
+    public boolean isSameTag(Tag otherTag) {
+        if (otherTag == this) {
+            return true;
+        }
+
+        return otherTag != null
+                && otherTag.getTagName().equals(getTagName());
+    }
+
+    public TagName getTagName() { return tagName; }
+
+    public int getNumberOfPersonsTagged() { return taggedCount; }
+
+    public void increaseTaggedCount() {
+        taggedCount++;
+    }
+
+    public void decreaseTaggedCount() {
+        taggedCount--;
+    }
+
+    /**
+     * Returns true if the tag can be deleted.
+     * The tag can be deleted if TaggedCount is 0.
+     */
```

Comment:
I know function is already quite obvious but not sure whether JavaDoc format requires a `@returns ...`

Reply:
I referenced this from JavaDocs from the Person class so I think it is okay?